src/test/java/com/limechain/network/protocol/grandpa/GrandpaServiceTest.java

```diff
@@ -79,6 +82,39 @@ void sendNeighbourMessageWhenExistingConnection() {
         }
     }
 
+    @Test
+    void sendCommitMessageWhenNotConnectionShouldSendHandshake() {
+        AddressBook addressBook = mock(AddressBook.class);
+        GrandpaController grandpaController = mock(GrandpaController.class);
+        when(connectionManager.getPeerInfo(peerId)).thenReturn(peerInfo);
+        when(peerInfo.getGrandpaStreams()).thenReturn(protocolStreams);
+        when(protocolStreams.getInitiator()).thenReturn(null);
+        when(host.getAddressBook()).thenReturn(addressBook);
+        when(protocol.dialPeer(host, peerId, addressBook)).thenReturn(grandpaController);
+
+        grandpaService.sendCommitMessage(host, peerId, encodedCommitMessage);
+
+        verify(grandpaController).sendHandshake();
+    }
+
+    @Test
+    void sendCommitMessageWhenExistingConnection() {
+        PeerInfo peerInfo = mock(PeerInfo.class);
+        ProtocolStreams protocolStreams = mock(ProtocolStreams.class);
+        Stream stream = mock(Stream.class);
```

Comment:
You don't have to mock these like that. I think 2 of them are already mocked with the @Mock annotation so you can add the 3rd one up there and delete these 3 rows.